### PR TITLE
Implement the JWK module and start on JWS

### PR DESCRIPTION
**Pull request by npmccallum**

### CODE REVIEW

File: jansson.pxd [hunk 1]
```diff
@@ -4,26 +4,9 @@ cdef extern from "jansson.h":
...
-        int position
-        char *source
-        char *text
-
-    json_t *json_pack(const char *fmt, ...)
-    json_t *json_pack_ex(json_error_t *error, size_t flags, const char *fmt, ...)
-
-    int json_unpack(json_t *root, const char *fmt, ...)
-    int json_unpack_ex(json_t *root, json_error_t *error, size_t flags, const char *fmt, ...)
+        pass
```
Comment (tiran) [inline jansson.pxd:8]: Please leave the fields here. I like to use them later to implement proper error reporting.

File: jose.pyx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import json
+
+def jwk_generate(jwk):
+    cdef jansson.json_t *cjwk = NULL
+    cdef char *ret = NULL
+
+    assert isinstance(jwk, dict)
+
+    try:
+        cjwk = jansson.json_loads(json.dumps(jwk).encode(u"UTF-8"), 0, NULL)
```
Comment (tiran) [inline jose.pyx:15]: PR #2 implements Python to Jansson json_t. May I suggest that we merge my PR first so you can use the feature in your PR?

File: jose.pyx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        assert cjwk
+
+        assert jose.jose_jwk_generate(cjwk)
```
Comment (tiran) [inline jose.pyx:18]: We shouldn't use asserts for error reporting.

File: jose.pyx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+
+def jwk_allowed(jwk, req=False, use=None, op=None):
+    cdef jansson.json_t *cjwk = NULL
+    cdef const char *cuse = NULL
+    cdef const char *cop = NULL
+
+    assert isinstance(jwk, dict)
+    assert op is None or isinstance(op, unicode)
+    assert use is None or isinstance(use, unicode)
```
Comment (tiran) [inline jose.pyx:59]: You can augment the function signature and let Cython do the error handling and type checking.

File: setup.py [hunk 5]
```diff
@@ -20,14 +20,14 @@
 ]
 
 setup(
-    name='pyjose',
+    name='jose',
```
Comment (tiran) [inline setup.py:23]: I don't like to compete with two other projects over the namespace. Let's find another name.

File: setup.py [hunk 6]
```diff
@@ -20,14 +20,14 @@
...
     description='Cython wrapper for libjose',
     ext_modules=cythonize(extensions),
     version='0.0.0',
     license='GPLv3+',
     maintainer='Latchset Contributors',
     maintainer_email='cheimes@redhat.com',
-    url='https://github.com/latchset/pyjose',
+    url='https://github.com/latchset/jose',
```
Comment (tiran) [inline setup.py]: The repository is pyjose.

**Review by tiran — CHANGES_REQUESTED** [review]

File: jansson.pxd (hunk 1 above)
Comment (npmccallum) [reply jansson.pxd:8]: There will never be error reporting from Jansson. We won't expose Jansson at all.

File: jose.pyx (hunk 2 above)
Comment (npmccallum) [reply jose.pyx:15]: I rejected PR #2. We aren't exposing Jansson.

File: jose.pyx (hunk 3 above)
Comment (npmccallum) [reply jose.pyx:18]: Let's use them for now just to get a rough version of the API working. Then you can see what the API should look like. Then I'm happy to change it once we have the big picture.

File: jose.pyx (hunk 4 above)
Comment (npmccallum) [reply jose.pyx:59]: +1. Can you merge this patch and then make that change as a separate commit?

File: setup.py (hunk 5 above)
Comment (npmccallum) [reply setup.py:23]: But we're wrapping jose. And the namespace is already collided. And none of the other packages are in a distro, so we'll be the first. We also implement the complete specs, which no other project does. I have aspirations to be the reference implementation. :)

File: setup.py (hunk 6 above)
Comment (npmccallum) [reply setup.py]: Can you merge this patch and just fix this one issue?

File: jose.pyx (hunk 2 above)
Comment (tiran) [reply jose.pyx:15]: We aren't going to use the json module either.

**Pull request merged**